plugins/controlboard/src/ControlBoardDriver.cpp

```diff
@@ -1144,13 +1159,120 @@ bool GazeboYarpControlBoardDriver::setPIDsForGroup_IMPEDANCE(std::vector<std::st
     }
 }
 
-bool GazeboYarpControlBoardDriver::setMinMaxImpedance()
+bool GazeboYarpControlBoardDriver::FindMotorControlGroup(yarp::os::Bottle& motorControlGroup_bot) const
 {
+    if (!m_pluginParameters.check("WRAPPER"))
+    {
+        yError()<<"Missing WRAPPER group";
+        return false;
+    }
+
+    if (!m_pluginParameters.findGroup("WRAPPER").check("networks"))
+    {
+        yError()<<"Missing networks group";
+        return false;
+    }
 
     yarp::os::Bottle& name_bot = m_pluginParameters.findGroup("WRAPPER").findGroup("networks");
     std::string name = name_bot.get(1).toString();
 
-    yarp::os::Bottle& kin_chain_bot = m_pluginParameters.findGroup(name);
+    motorControlGroup_bot = m_pluginParameters.findGroup(name);
+    return true;
+}
+
+bool GazeboYarpControlBoardDriver::setPositionsToleranceLinear()
+{
+    yarp::os::Bottle kin_chain_bot;
+    if (!FindMotorControlGroup(kin_chain_bot))
+        return false;
+
+    if (!kin_chain_bot.check("positionToleranceLinear")) {
+        yWarning()<<"No positionToleranceLinear value found in ini file, default one will be used!";
+        yDebug()<<"positionToleranceLinear: [ "<<m_robotPositionToleranceLinear<<" ]";
+        return true;
+    }
+
+    yInfo()<<"positionToleranceLinear param found!";
+    yarp::os::Bottle& positionToleranceLinear_bot = kin_chain_bot.findGroup("positionToleranceLinear");
+    if (static_cast<size_t>(positionToleranceLinear_bot.size()) != 2) {
+        yError()<<"Invalid number of params:positionToleranceLinear:"<<static_cast<size_t>(positionToleranceLinear_bot.size());
+        return false;
+    }
+
+    auto tmp=positionToleranceLinear_bot.get(1);
```

Comment:
I think here we can replace auto with `yarp::os::Value` for the sake of readability.

Reply:
Ok